text/0043-block-metadata.md

```diff
@@ -0,0 +1,359 @@
+      }
+    }
+  ]
+}
+```
+
+</details>
+
+## Block Version bump
+
+Block version 3 (release v1.3) will make the `block_metadata` required, at which
+point consensus nodes and clients should reject blocks without metadata.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- This makes each block bigger (more bytes), but we gain an important ability to
+  backfill a ledger from `ArchiveBlocks`.
+- A introduces complexity around backfilling a ledger, namely looking up
+  historical AVRs.
+- Also entails additional configuration, with the associated operational cost.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+This addition of AVRs allows consumers to verify the blockchain with fewer data
+sources, including post-hoc validation that the blocks were externalized by
+legitimate consensus enclaves. This enables a single source of truth in the
+steady state (rather than requiring a Watcher DB in addition to the actual block
+data).
+
+While this capability is necessary for block streaming, it is useful to decouple
+the step of adding AVRs and quorum sets to the blockchain.
+
+## Alternatives considered
+
+#### Write metadata for older blocks
+
+It would be nice if we can generate metadata for older blocks, as that would
+keep the lookup logic simple, but we cannot generate signatures since MCF does
+not have all the private signing keys, nor should it.
+
+# Prior art
+[prior-art]: #prior-art
+
+The author has not found any direct analogs for including network metadata in a
+blockchain.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+None as of this writing.
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+1. Enables the streaming and archive APIs defined in
+   [MCIP #29](0029-block-streaming.md) to use equivalent representations across
+   the streaming and archive block APIs.
+
+1. Enables block streaming and other consumers to include and validate the AVRs
+   for persisted blocks, which opens a path to replacing mobilecoind and watcher
+   in Fog.
+
+   Note that block timestamps come from the block signature, and nodes that are
+   catching up to the latest consensus (including mobilecoind, full-service, and
+   consensus nodes that are catching up) do not append any block signature,
+   relying on a Watcher to update the signatures. Replacing Watcher thus entails
+   updating this catch-up mode to handle block signatures as well as metadata,
+   perhaps reusing SCP.
```

Comment:
```suggestion
   relying on a Watcher to provide timestamps (by syncing all blocks from multiple sources,
   at least one is guaranteed to contain a signature and a timestamp).
   Replacing Watcher thus entails updating this catch-up mode to handle block signatures as well
   as metadata, perhaps reusing SCP quorum sets.
```

(we don't store SCP messages so that is not reusable, but perhaps a node can choose to accept a signature(+timestamp) if a quorum set check can be satisfied after having fetched the same block from multiple sources. Either way that is out of scope for this MCIP)